src/operations/indexes.ts

```diff
@@ -458,39 +419,31 @@ export class IndexExistsOperation extends AbstractCallbackOperation<boolean> {
     this.indexes = indexes;
   }
 
-  override executeCallback(
-    server: Server,
-    session: ClientSession | undefined,
-    callback: Callback<boolean>
-  ): void {
+  override async execute(server: Server, session: ClientSession | undefined): Promise<boolean> {
     const coll = this.collection;
     const indexes = this.indexes;
 
-    indexInformation(
-      coll.s.db,
-      coll.collectionName,
-      { ...this.options, readPreference: this.readPreference, session },
-      (err, indexInformation) => {
-        // If we have an error return
-        if (err != null) return callback(err);
-        // Let's check for the index names
-        if (!Array.isArray(indexes)) return callback(undefined, indexInformation[indexes] != null);
-        // Check in list of indexes
-        for (let i = 0; i < indexes.length; i++) {
-          if (indexInformation[indexes[i]] == null) {
-            return callback(undefined, false);
-          }
-        }
-
-        // All keys found return true
-        return callback(undefined, true);
+    const info = await indexInformation(coll.s.db, coll.collectionName, {
+      ...this.options,
+      readPreference: this.readPreference,
+      session
+    });
+    // Let's check for the index names
+    if (!Array.isArray(indexes)) return info[indexes] != null;
+    // Check in list of indexes
+    for (let i = 0; i < indexes.length; i++) {
+      if (info[indexes[i]] == null) {
+        return false;
       }
-    );
+    }
+
+    // All keys found return true
+    return true;
```

Comment:
(optional cleanup)

We could clean this up a bit if we checked in the constructor if `indexes` was an array and if it wasn't, converting it to a string.  Then we only handle the case where `indexes` is an array.  We can also use a for-of loop while we're here.

Another option would be to use array builtins (instead of the loop entirely):
```
return Array.every(indexName => info[indexName] != null)
```

or a third option would be to use `full: true` and sets:

```
const indexes = await indexInformation(coll.s.db, coll.collectionName, {
      ...this.options,
      readPreference: this.readPreference,
      session,
	  full: true
    });

const names = indexes.map(({ name }) => name);

// isSuperSet is defined in util.ts
return isSuperSet(names, this.indexes);

////// or we get rid of indexInformation here entirely 
const indexesOnCollection = this.collection
    .listIndexes({ 
        ...this.options, 
        readPreference: this.readPreference, session })
    .map(({ name }) => name)
    .toArray();

return isSuperSet(indexesOnCollection, this.indexes);
```


feel free to refactor this if you want but this is optional

Reply:
I used the second option I think it's the cleanest